src/clarity-angular/data/datagrid/providers/selection.ts

```diff
@@ -156,9 +156,14 @@ export class Selection {
      */
     public isSelected(item: any): boolean {
         if (this._selectionType === SelectionType.Single) {
-            return this.currentSingle === item;
+            return (this.currentSingle)
+                ? this._items.trackBy(0, this.currentSingle) === this._items.trackBy(0, item)
```

Comment:
My previous comment becomes immediately obvious here, because we don't have an index so we use `0` instead of a wrong index. But the problem is the same, neither works for index-dependent trackBy functions.

To test this, the easiest is to use a track by index: `(index: number, item: any) => index`.
I believe in your server-driven case if I use this trackBy function, select one item on the first page, then move to the second page, every item on the new page will be selected.